phonon/client.py

```diff
@@ -0,0 +1,42 @@
+import redis
+import zlib
+
+
+class ShardedClient(object):
+
+    key_mappings = {
+        'get': 0, 'hget': 0,
+        'set': 0, 'hset': 0,
+        'hgetall': 0,
+    }
+
+    def __init__(self, hosts=None, port=6379, db=0):
+        self.hosts = sorted(hosts)
+        self.clients = [redis.StrictRedis(host=host, port=port, db=db) for host in self.hosts]
+
+    def get_key(self, method, *args, **kwargs):
+        try:
+            return args[self.key_mappings[method]]
+        except KeyError, e:
+            raise NotImplemented("You should implement (or fix?) a key mapping for {}".format(method))
+
+    def route(self, key):
+        return self.clients[zlib.crc32(key) % len(self.clients)]
```

Comment:
From [the docs][1]:

> **Note** To generate the same numeric value across all Python versions and platforms use `crc32(data) & 0xffffffff`. If you are only using the checksum in packed binary format this is not necessary as the return value is the correct 32bit binary representation regardless of sign.

[1]: https://docs.python.org/2/library/zlib.html#zlib.crc32